js/hoot/control/utilities/ImportDirectory.js

```diff
@@ -880,6 +946,38 @@ Hoot.control.utilities.importdirectory = function(context) {
         return true;
     };
 
+    /**
+    * @desc Function to append FCode Description based on FCODE
+    * @desc Assumes that layer name is a valid FCODE
+    * @oaram fcodeList - JSON of FCODEs and Descriptions
+    * @param filesList - Selected files list.
+    **/
+    var appendFCodeDescription = function(fcodeList) {
+        var filesList = _getFilesList();
+        _.each(filesList, function(f){
+            var fcodeMatch = _.find(this,{name:f.value}) || _.find(this,{fcode:f.value});
+            if(fcodeMatch){
+                var fcodeName = f.value + '_' + fcodeMatch.desc.replace(' ','_');
+                // Remove any special characters
+                fcodeName = context.hoot().removeSpecialChar(fcodeName);
+                var selOpt = d3.select('#importDirectoryFilesList').select('option[value="' + f.value + '"]');
+                selOpt.text(fcodeName);
+            }
+        },fcodeList);
+        _validateFileList(_getFilesList());
+    };
+
+    var removeFCodeDescription = function() {
+        // Replace text of each option with value (original value)
+        var opts = d3.select('#importDirectoryFilesList').selectAll('option')[0];
+        _.each(opts,function(opt){
```

Comment:
I'm bothered by the use of [0] here and then lodash and other array methods when I think there is probably a d3 selection method that will do the same thing.  It feels hacky.